modules/plottinggl/include/modules/plottinggl/utils/axisrenderer.h

```diff
@@ -51,90 +53,245 @@ class Texture2D;
 
 namespace plot {
 
+namespace detail {
+
+struct Resetter {
```

Comment:
this look like a quite complex solution. Some comments would be helpful if we ever want to change this...
I guess it is used to keep things in sync when properties change?